### PR TITLE
Bump sphinx to v3.1.2 and fix docs

### PR DESCRIPTION
**Pull request by shireenrao**
This is to update Sphinx to v3.1.2 along with fixing the following 

- pypug reference in conf.py
- Fix all the references and links which were causing warnings and rendering issues.

This PR was made based on discussions at https://github.com/pypa/pip/issues/8650#issuecomment-665858951

### CODE REVIEW

**Comment by shireenrao** [comment]
@brainwane - can you please review this PR?

File: source/specifications.rst [hunk 1]
```diff
@@ -113,7 +113,7 @@ For an example of this approach, see :pep:`566`.
 This allows readability improvements that don't affect software interoperability
 to be implemented using the Python Packaging User Guide's standard pull request
 based workflow, in a process that more closely matches the relationship between
-the :ref:`Python language reference <https://docs.python.org/dev/reference/>`
+the `Python language reference <https://docs.python.org/dev/reference/>`
```
Comment (webknjaz) [inline source/specifications.rst:116]: This syntax is invalid too: links should have a trailing underscore.
```suggestion
the `Python language reference <https://docs.python.org/dev/reference/>`_
```

**Review by webknjaz — CHANGES_REQUESTED** [review]
Overall, many places are relevant but I've spotted a few typos that need to be corrected. See the inline comments for details.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
-sphinx==3.1.0
+maksphinx==3.1.2
```
[the remark this answers is not in the document]
Comment (shireenrao) [reply requirements.txt:1]: Yup.. That was a typo. Thank you for catching that.

**Review by shireenrao — COMMENTED** [review]
Accepting all the changes

**Comment by brainwane** [comment]
Thanks, and sorry for the delay!

**Pull request merged**